### ISSUE TITLE
Media Library Pro stopped working after upgrading to v3

### ISSUE THREAD

**Issue by amshehzad** [spatie/medialibrary.pro#54]
I've followed the upgrade guide but I'm getting this error:

![image](https://github.com/spatie/medialibrary.pro/assets/15146578/e02d9485-40f2-47d0-b851-88fa9fe401b0)

**Comment by riasvdv** [comment]
Can you share the full exception with us? I'm lacking a bit of context with just the screenshot

**Comment by amshehzad** [comment]
Yes sure here is full error stack:
https://flareapp.io/share/NPL8lbQP

**Comment by amshehzad** [comment]
Sorry my bad, I missed this part from upgrade:

> MAKE SURE YOU'RE SETTING YOUR MEDIA PROPERTIES TO A DEFAULT EMPTY ARRAY

I changed  `public array $documents;` to `public array $documents = [];` 
and the problem is solved.

but I'm still not able to use the component. in console I can see a lot of errors:
![image](https://github.com/spatie/medialibrary.pro/assets/15146578/ad469ba5-6362-443a-8b9c-e61e77fb39f4)

If I try to upload something despite these errors I get this error:
https://flareapp.io/share/W7zDk6lm

**Comment by riasvdv** [comment]
Can you share your full console errors including the warnings?

**Comment by amshehzad** [comment]
Yes here it is:
![image](https://github.com/spatie/medialibrary.pro/assets/15146578/f100ba3c-0b79-417c-abfe-c58fe833ff33)

![image](https://github.com/spatie/medialibrary.pro/assets/15146578/2e79941b-bcf5-4639-a6f6-3c0c319b82ad)

![image](https://github.com/spatie/medialibrary.pro/assets/15146578/fc2d1de3-ae3d-48e8-bbce-f4d5f8e616a1)

![image](https://github.com/spatie/medialibrary.pro/assets/15146578/5f2c64e9-8b81-4775-997e-69f9776bf175)

**Comment by riasvdv** [comment]
Can you try running `php artisan view:clear` and see if that solves your issue?

**Comment by amshehzad** [comment]
Unfortunatly it didn't

**Comment by amshehzad** [comment]
if it's possible, can we connect on discord so I can share more details or share whatever you need to check more details about the issue?

**Comment by amshehzad** [comment]
Ok I created a new repo but the error is not there, after some debugging it seems like the problem occurs when I use it inside wire-elments modal component. I'll try to setup a new repo with wire-elements producing the issue

**Comment by amshehzad** [comment]
@riasvdv 
please check the repo:
https://github.com/amshehzad/medialibrary

the component does not work inside wire-elements modal

**Comment by amshehzad** [comment]
@riasvdv I've discussed the issue with Philo the creator of wire-elements and here is what he suggested: 

I found that the issue is related to how Livewire works.
The modals are loaded dynamically and the <script> tag is not being executed for this reason:

vendor/spatie/laravel-medialibrary-pro/resources/views/livewire/uploader.blade.php

What seems to work is to copy the <script> tag contents, make one little tweak (it uses the {{ }} blade tags for translations. So If I add the following to my app.js it works:

https://gist.github.com/PhiloNL/54727f23d2210ed0979cb946f06e528a

This has fixed the issue but now I've run into a new issue. when the media component initializes, I get this error:

![image](https://github.com/spatie/medialibrary.pro/assets/15146578/8793e98b-a9e5-4634-9fb4-b4d4e397bd07)

https://flareapp.io/share/x5MEGRb5

Upon debugging, I found that the media component calls a method named the same as the media property ( in my case its documents ) I don't understand why this call is being made: 
![image](https://github.com/spatie/medialibrary.pro/assets/15146578/8c834bf9-26f9-4f30-b2f9-1124440a067b)
